File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& a) {
        
        int maxn=INT_MIN;
        int n=a.size();
        int s=0;
        for(int i=0;i<n;i++){
            s=s+a[i];
            maxn=max(maxn,s);
            if(s<0){
                s=0;
            }
        }
        return maxn;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] a) {
        int maxn=Integer.MIN_VALUE;
        int sum=0,n=a.length;

        for(int i=0;i<n;i++){
            sum+=a[i];
            maxn=Math.max(sum,maxn);
            if(sum<0){
                sum=0;
            }

        }
        return maxn;
    }
}